### ISSUE TITLE
does not do anything on windows 11 24h2

### ISSUE THREAD

**Issue by doxycomp** [Aldaviva/AuthenticatorChooser#2]
Build 26100.712

Visible in Taskmanager, but does not do anything when the sign in window appears.

**Comment by Aldaviva** [comment]
What language/locale is Windows set to? Also, is this Windows x64 or ARM64?

Thanks.

**Comment by doxycomp** [comment]
Hi that was quick :)

my locale is:
Systemgebietsschema:                           de;Deutsch (Deutschland)
Eingabegebietsschema:                          de;Deutsch (Deutschland)

and it is a Windows 11 pro x64 on Intel.

Let me know if I can provide further information!

Kind regards

**Comment by Aldaviva** [comment]
Thanks, I will try to replicate this issue soon and figure out what is going on.

Has this program ever worked for you in the past on previous builds of Windows 11 (like 23H2)?

**Comment by doxycomp** [comment]
Thanks a lot :)

Sorry, I just found out about your App today and gave it a try on my current system. So I cannot provide that information.
My other two systems do not have a bluetooth adapter, so I could not replicate the issue there :/

**Comment by Aldaviva** [comment]
Not a problem, it's just useful to narrow down that it may be more likely a localization issue than a Windows update version issue.

If you like, you can attach a screenshot of the Windows Security dialog box with the two phone and USB choices visible. If not, I will see it when I set up a VM with the same locale and replicate it there.

**Comment by doxycomp** [comment]
Sure thing!

here is the screenshot.
I also gave it a shot on my colleagues notebook with the same behavior. Same windows version, but we had to install .net 8.

![2024-06-19 17 46 54 security box](https://github.com/Aldaviva/AuthenticatorChooser/assets/38212682/6d131935-e34a-4b5e-97a2-918d39a17c8a)

**Comment by Aldaviva** [comment]
My guess now is that Windows 11 24H2 has started localizing some text that was previously always English in this dialog box in 23H2, so my program can't find it now that the text has changed. I will need to install 24H2 and all language packs to extract the new localized strings. Unfortunately, it takes forever (several days) to install all Windows language packs, but at least I learned how to automate it. I will try to send you an unofficial development snapshot build sooner to test out and see if that fixes it.

**Comment by doxycomp** [comment]
Thanks a lot,

I don't know if this helps: I just changed my region settings to English (USA) and gave it another try without success.
Perhaps we do not have to waste your time on localization :)
The text looked like that:
![Screenshot 2024-06-19 180325 security box](https://github.com/Aldaviva/AuthenticatorChooser/assets/38212682/5f5fd09e-b440-4710-b655-d878eacf0df2)

**Comment by Aldaviva** [comment]
Hmm interesting. Thanks for that tip. That's the same text that 23H2 used to show for everyone, so there may be multiple problems here, localization changes and dialog structure changes.

If you want, you can send me the file that contains your localized strings for this dialog box: `C:\Windows\System32\de-DE\fidocredprov.dll.mui`. You will probably have to ZIP this file so GitHub will allow you to upload it. Once we can see where the strings are coming from, we can adapt it to dynamically support all languages.

**Comment by doxycomp** [comment]
No problem at all :)

[fidocredprov.dll.mui.zip](https://github.com/user-attachments/files/15903725/fidocredprov.dll.mui.zip)


I was also able to reproduce the problem on windows 11 22h2 22621.3296 on german locale on another notebook in the mean time.

**Comment by Aldaviva** [comment]
Here is a [development build](https://github.com/user-attachments/files/15908919/AuthenticatorChooser.zip).
- It should be compatible with **Windows 23H2** in German, as well as all other languages.
    - I fixed a bug that was not finding some of the buttons correctly based on their localized names (unexpected suffix).
    - I also fixed a bug that was preventing it from automatically detecting the localization for other languages at runtime (DLL reading was incorrectly hardcoded to always English).
- It might be compatible with **Windows 24H2** in German.
    - I could not find the new dialog heading text in `fidocredprov.dll.mui`, so it must have been added to a different file. I will have to join the Insider program and install a Windows 24H2 Release Preview build to search for it.
    - This won't support other languages besides German and English because I manually added the text from your screenshot.

I am still not sure why it failed for you on 24H2 in English. Maybe one of these other fixes will cover it.

**Comment by doxycomp** [comment]
Thank you so much for putting in all the effort, it works like a charm now on German Win 11 24H2.

It is so much more convenient this way :)

Wich kind regards!

**Comment by Aldaviva** [comment]
I have figured out more about why this wasn't working before. Aside from all the bugs I fixed, I also didn't understand that I needed to change the Windows OS system-wide locale in addition to the Windows user locale. That's why, when I was testing with  a Fresh user account on an English system, I saw a half-French, half-English dialog box. Now that I set the system-wide locale to German I see the localized title that you saw too. I think this is caused by Windows getting some of the strings in this dialog box from the CryptSvc service (which runs as the system) and some of them from the CredentialUIBroker process (which runs as the user), so the localization is split this way.

In my new Windows 11 24H2 Insider Release Preview VM, I have tracked down the strings in there to a different file, `webauthn.dll.mui`. I have updated this program to read the localized strings from that file at runtime, and I have also copied all languages of this new DLL file. The next step is to update my program that bulk extracts strings from DLL files and converts them into .NET RESX files that can be compiled into this program, which needs to be updated to support extracting from more than one file of the same language.
